bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -355,7 +355,7 @@ private static boolean isConvertibleToRecordType(Object sourceValue, BRecordType
             } else {
                 if (targetFieldTypes.containsKey(fieldName)) {
                     if (getConvertibleTypes(valueEntry.getValue(), targetFieldTypes.get(fieldName),
-                            unresolvedValues).size() != 1) {
+                            unresolvedValues).size() == 0) {
```

Comment:
Don't we still have to check for > 1 convertible types? That should also be an error, right? Let's add a test for the same.

The following should fail.

```ballerina
public type Maps record {|int i; int...;|}|record {|int i?;|};

public type Value record {|
    Maps value;
|};

public function main() {
    json j = {"value": <map<int>> {i: 1}};
    Value res = checkpanic j.cloneWithType(Value);
}
```